src/components/Form/Form.tsx

```diff
@@ -0,0 +1,151 @@
+import "./style.scss";
+import { useFormik } from "formik";
+import { Modal } from "../../interfaces";
+import { films } from "../../films";
+import { CONSTANTS } from "../../constants";
+import Select from "react-select";
+import DeleteForm from "./DeleteForm";
+
+const selectStyle: any = {
+  menu: () => {
+    return {
+      backgroundColor: "#232323;",
+      position: "absolute",
+      margin: "20px",
+      marginTop: "0",
+      display: "block",
+      width: "330px",
+      top: "48px",
+      borderRadius: "3px",
+    };
+  },
+  option: () => {
+    return {
+      ":hover": {
+        backgroundColor: "#f65261;",
+      },
+    };
+  },
+  control: () => {
+    return {
+      backgroundColor: "#555",
+      margin: "20px",
+      border: "none",
+      height: "26px",
+      borderRadius: "3px",
+      color: "white",
+      display: "flex",
+    };
+  },
+};
+
+interface FormProps {
+  closeModal(): void;
+  modalState: Modal;
+}
+
+const Form: React.FC<FormProps> = ({ closeModal, modalState }) => {
+  const movie: any =
+    modalState.type === CONSTANTS.FORM_TYPE.ADD
+      ? CONSTANTS.ADD_MOVIE
+      : films.find((el) => {
+          return el.id === modalState.filmID;
+        });
+
+  const formik = useFormik({
+    initialValues: movie,
+    onSubmit: (values) => {
+      values.releaseDate = new Date(values.releaseDate);
+      if (modalState.type === CONSTANTS.FORM_TYPE.EDIT) {
+        films[films.indexOf(movie)] = values;
+      } else if (modalState.type === CONSTANTS.FORM_TYPE.DELETE) {
+        films.splice(films.indexOf(movie), films.indexOf(movie));
+      } else {
+        films.push(values);
+      }
```

Comment:
not the best way to manage everything in one place with if else (if there is more then 2 if). If more if cases and forms appear in that component  it will become harder to read, maintain, and edit all this logic.
it is better to divide logic and create separate form at least for `DELETE`.